In [1]:
# Import Required Libraries
import pandas as pd
import numpy as np

# Term Frequencies (TF) for Doc1, Doc2, Doc3
tf_data = {
    'term': ['car', 'auto', 'insurance', 'best'],
    'Doc1': [27, 3, 0, 14],
    'Doc2': [4, 33, 33, 0],
    'Doc3': [24, 0, 29, 17]
}

# IDF Values
idf_data = {
    'term': ['car', 'auto', 'insurance', 'best'],
    'idf': [1.65, 2.08, 1.62, 1.5]
}

# Create DataFrames
tf_df = pd.DataFrame(tf_data)
idf_df = pd.DataFrame(idf_data)

# Merge TF and IDF DataFrames
data = pd.merge(tf_df, idf_df, on='term')

# Compute TF-IDF
for doc in ['Doc1', 'Doc2', 'Doc3']:
    data[f'TF-IDF_{doc}'] = data[doc] * data['idf']

# Display Results
print(data[['term', 'TF-IDF_Doc1', 'TF-IDF_Doc2', 'TF-IDF_Doc3']])

        term  TF-IDF_Doc1  TF-IDF_Doc2  TF-IDF_Doc3
0        car        44.55         6.60        39.60
1       auto         6.24        68.64         0.00
2  insurance         0.00        53.46        46.98
3       best        21.00         0.00        25.50


# Results

The table below shows the computed TF-IDF weights for each term in each document:

| Term      | TF-IDF Doc1 | TF-IDF Doc2 | TF-IDF Doc3 |
| --------- | ----------- | ----------- | ----------- |
| car       | 44.55       | 6.6         | 39.6        |
| auto      | 6.24        | 68.64       | 0.0         |
| insurance | 0.0         | 53.46       | 46.98       |
| best      | 21.0        | 0.0         | 25.5        |


In [2]:
# Verify Results
# Check specific TF-IDF values
print("TF-IDF for 'car' in Doc1:", 27 * 1.65)
print("TF-IDF for 'auto' in Doc2:", 33 * 2.08)
print("TF-IDF for 'insurance' in Doc3:", 29 * 1.62)
print("TF-IDF for 'best' in Doc1:", 14 * 1.5)

TF-IDF for 'car' in Doc1: 44.55
TF-IDF for 'auto' in Doc2: 68.64
TF-IDF for 'insurance' in Doc3: 46.980000000000004
TF-IDF for 'best' in Doc1: 21.0
